tests/asynch/basic.rs

```diff
@@ -311,3 +311,58 @@ fn drop_shuttle_yield_future() {
     assert_eq!(2, async_accesses_clone.load(Ordering::SeqCst));
     assert_eq!(1, post_yield_accesses_clone.load(Ordering::SeqCst));
 }
+
+// This test checks two behaviors. First, it checks that a future can poll another future
+// and wake itself in the same poll() call. Second, it checks that a waiter can finish
+// before the waitee task.
```

Comment:
If I understand right, the first behavior is actually that a task can be polled twice without needing to wake up any task waiting on it in between.

Reply:
Yes, exactly. I'll change the comment to something like this because maybe there are other possible causes besides this specific scenario.